skbio/sequence/_sequence.py

```diff
@@ -106,20 +99,181 @@ class Sequence(collections.Sequence, SkbioObject):
+
+    >>> pprint(metadata)
+    {'authors': ['Alice'], 'desc': 'seq desc', 'id': 'seq-id'}
+    >>> seq.metadata == metadata
+    False
+
+    Note however that since only a *shallow* copy was made, updates to mutable
+    objects will also change the original metadata dictionary:
+
+    >>> seq.metadata['authors'].append('Bob')
+    >>> seq.metadata['authors']
+    ['Alice', 'Bob']
+    >>> metadata['authors']
+    ['Alice', 'Bob']
+
+    This behavior can also occur when manipulating a sequence that has been
+    derived from another sequence:
+
+    >>> subseq = seq[1:3]
+    >>> subseq
+    Sequence('CG', length=2, has_metadata=True, has_positional_metadata=False)
+    >>> pprint(subseq.metadata)
+    {'authors': ['Alice', 'Bob'],
+     'desc': 'seq desc',
+     'id': 'new-id',
+     'pubmed': 12345}
+
+    The subsequence has inherited the metadata of its parent sequence. If we
+    update the subsequence's author list, we see the changes propagated in the
+    parent sequence and original metadata dictionary:
+
+    >>> subseq.metadata['authors'].append('Carol')
+    >>> subseq.metadata['authors']
+    ['Alice', 'Bob', 'Carol']
+    >>> seq.metadata['authors']
+    ['Alice', 'Bob', 'Carol']
+    >>> metadata['authors']
+    ['Alice', 'Bob', 'Carol']
+
+    The behavior for updating positional metadata is similar. Let's create a
+    new sequence with positional metadata that is already stored in a
+    ``pd.DataFrame``:
+
+    >>> positional_metadata = pd.DataFrame(
+    ...     {'qual': [3, 3, 4, 10], 'list': [[], [], [], []]})
+    >>> seq = Sequence('ACGT', positional_metadata=positional_metadata)
+    >>> seq # doctest: +NORMALIZE_WHITESPACE
+    Sequence('ACGT', length=4, has_metadata=False,
+             has_positional_metadata=True)
+    >>> seq.positional_metadata
+      list  qual
+    0   []     3
+    1   []     3
+    2   []     4
+    3   []    10
+
+    Now let's update the sequence's positional metadata by adding a new column
+    and changing a value in another column:
+
+    >>> seq.positional_metadata['gaps'] = [False, False, False, False]
+    >>> seq.positional_metadata.loc[0, 'qual'] = 999
+    >>> seq.positional_metadata
+      list  qual   gaps
+    0   []   999  False
+    1   []     3  False
+    2   []     4  False
+    3   []    10  False
+
+    Note that the original positional metadata (stored in variable
+    ``positional_metadata``) hasn't changed because a shallow copy was made:
+
+    >>> positional_metadata
+      list  qual
+    0   []     3
+    1   []     3
+    2   []     4
+    3   []    10
+    >>> seq.positional_metadata.equals(positional_metadata)
+    False
+
+    Next let's create a sequence that has been derived from another sequence:
+
+    >>> subseq = seq[1:3]
+    >>> subseq
+    Sequence('CG', length=2, has_metadata=False, has_positional_metadata=True)
+    >>> subseq.positional_metadata
+      list  qual   gaps
+    0   []     3  False
+    1   []     4  False
```

Comment:
I think there should be an example of setting a non-mutable piece of metadata to make it clear that only mutable objects have this behavior, simple things like ints and strings are fine.